src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/InfrastructureDefinition/__tests__/BootstrapDeployInfraDefinition.test.tsx

```diff
@@ -383,4 +422,71 @@ describe('BootstrapDeployInfraDefinition tests', () => {
     expect(queryByText(serviceSelectPortalDiv, 'orgLabel')).toBeInTheDocument()
     expect(queryByText(serviceSelectPortalDiv, 'projectLabel')).not.toBeInTheDocument()
   })
+
+  test('it should allow services from all scopes for scoping if project level infra is selected', async () => {
+    const ref = React.createRef<InfraDefinitionWrapperRef>()
+
+    const { container } = render(
+      <TestWrapper
+        path={routes.toEnvironmentDetails({
+          ...accountPathProps,
+          ...orgPathProps,
+          ...projectPathProps,
+          ...environmentPathProps
+        })}
+        pathParams={{
+          accountId: 'testAcc',
+          orgIdentifier: 'testOrg',
+          projectIdentifier: 'testProject',
+          environmentIdentifier: 'testEnv'
+        }}
+        defaultFeatureFlagValues={{ CDS_SCOPE_INFRA_TO_SERVICES: true, NG_SVC_ENV_REDESIGN: true, CDP_AWS_SAM: true }}
+      >
+        <BootstrapDeployInfraDefinitionWithRef
+          closeInfraDefinitionDetails={jest.fn()}
+          refetch={refetchInfrastructureList}
+          environmentIdentifier={'env1'}
+          isReadOnly={false}
+          scope={Scope.PROJECT}
+          handleInfrastructureUpdate={handleInfrastructureUpdate}
+          isInfraUpdated={false}
+          isSingleEnv={true}
+          ref={ref}
+          selectedInfrastructure="abc"
+          infrastructureDefinition={{
+            identifier: 'abc',
+            orgIdentifier: 'testOrg',
+            projectIdentifier: 'testProject',
+            name: 'Project Level Infra',
+            deploymentType: ServiceDeploymentType.AwsSam,
+            type: 'AWS_SAM',
+            spec: {
+              connectorRef: '',
+              region: ''
+            }
+          }}
+        />
+      </TestWrapper>
+    )
+
+    const queryByNameAttribute = (name: string): HTMLElement | null => queryByAttribute('name', container, name)
+
+    const portalDivs = document.getElementsByClassName('bp3-portal')
+    expect(portalDivs.length).toBe(0)
+
+    const scopeToSpecificServicesCheckbox = queryByNameAttribute('scopeToSpecificServices')
+    expect(scopeToSpecificServicesCheckbox).toBeInTheDocument()
+    await userEvent.click(scopeToSpecificServicesCheckbox!)
+    await waitFor(() => expect(scopeToSpecificServicesCheckbox).toBeChecked())
+
+    const scopedServicesInput = await screen.findByText('cd.pipelineSteps.serviceTab.selectServices')
+    await waitFor(() => expect(scopedServicesInput).toBeInTheDocument())
```

Comment:
This can be simplified to this:

```suggestion
    await findByText('cd.pipelineSteps.serviceTab.selectServices')
```

It also waits for this text to be available, and fails the test if not found within a timeout.

Reply:
Makes sense, @abhinavrastogi-harness . Looks like I ended up over-simplifying. 😄 
I will keep this in mind.